flow/connectors/postgres/qrep.go

```diff
@@ -264,6 +264,13 @@ func (c *PostgresConnector) getNumRowsPartitions(
 
 		return partitionHelper.GetPartitions(), nil
 	} else {
+		// Special handling for CTID when user wants a fixed number of partitions:
+		// build approximate physical (block-based) TID partitions without sorting or MIN/MAX(ctid).
+		if strings.EqualFold(config.WatermarkColumn, "ctid") {
```

Comment:
```suggestion
		if config.WatermarkColumn == "ctid" {
```
only lowercase ctid is special